### DISCUSSION TITLE
/usr/local/opt not owned by logged in user when installing from package on Intel Macs

### DISCUSSION THREAD

**Discussion by fraserhess** [Homebrew/discussions#5052] [Getting started] [answered]
### Output of `brew config`

```shell
% brew config   
HOMEBREW_VERSION: 4.2.3
ORIGIN: https://github.com/Homebrew/brew
HEAD: b3751bca8c4a3c76107d5aa3b75b81db93cf4bb6
Last commit: 5 days ago
Core tap JSON: 08 Jan 13:55 UTC
Core cask tap JSON: 08 Jan 13:55 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.1.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: octa-core 64-bit icelake
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 13.6.3-x86_64
CLT: 15.1.0.0.1.1700200546
Xcode: N/A
```

### Output of `brew doctor`

```shell
% brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: The following directories are not writable by your user:
/usr/local/opt

You should change the ownership of these directories to your user.
  sudo chown -R macadmin /usr/local/opt

And make sure that your user has write permission.
  chmod u+w /usr/local/opt
```

### Description of issue

After installing from package on an Intel Mac, `brew install` commands error because the user does not own /usr/local/opt

Install the Homebrew 4.2.3 package on an Intel Mac.
The postinstall script creates /usr/local/opt on line 80 https://github.com/Homebrew/brew/blob/302e83e9aaa0a3a4c0644f43a0302a582128ed1f/package/scripts/postinstall#L80
However, it does not grant ownership of that directory on line 95 https://github.com/Homebrew/brew/blob/302e83e9aaa0a3a4c0644f43a0302a582128ed1f/package/scripts/postinstall#L95
Running for example, `brew install jq`, results in an error
```
% brew install jq
Error: The following directories are not writable by your user:
/usr/local/opt

You should change the ownership of these directories to your user.
  sudo chown -R macadmin /usr/local/opt

And make sure that your user has write permission.
  chmod u+w /usr/local/opt
```
Following the guidance in the error message resolves the issue

**Comment by MikeMcQuaid** [comment]
> However, it does not grant ownership of that directory on line 95

Could you try and open a pull request? [This document](https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request) should help and we're happy to walk you through anything else.

Thanks!

**Reply by MikeMcQuaid** [reply, marked as answer]
Ignore me: I just merged a PR that did this: https://github.com/Homebrew/brew/pull/16480

**Reply by fraserhess** [reply]
This has resolved the issue
